### PR TITLE
Patch rook-ceph-tools deployment on consumer

### PR DESCRIPTION
**Pull request by jilju**
Patch rook-ceph-tools deployment on consumer to run ceph commands. The patch is done after deployment.
Signed-off-by: Jilju Joy <jijoy@redhat.com>

### CODE REVIEW

**Comment by jilju** [comment]
Verified: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/11993//consoleFull

```
......
2022-04-21 20:05:29  14:35:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage patch Deployment rook-ceph-tools -n openshift-storage -p '[{"op": "replace", "path": "/spec/template/spec/containers/0/env", "value":[{"name": "ROOK_CEPH_USERNAME", "value": "client.admin"}, {"name": "ROOK_CEPH_SECRET", "value": "AQDlPWFiDymyGxAAJy1Pee0HpTFRyMHijkRAGw=="}]}]' --type json

........

2022-04-21 20:06:07  14:36:07 - MainThread - ocs_ci.helpers.helpers - INFO - Pod rook-ceph-tools-57b5c6bc64-fsx7n reached state Running
2022-04-21 20:06:07  14:36:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
2022-04-21 20:06:07  14:36:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
2022-04-21 20:06:07  14:36:07 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-57b5c6bc64-fsx7n -- ceph health
2022-04-21 20:06:08  14:36:08 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
.....
```

File: ocs_ci/deployment/rosa.py [hunk 1]
```diff
@@ -147,24 +147,29 @@ def deploy_ocs(self):
...
-            selector="app=rook-ceph-osd",
-            resource_count=3,
-            timeout=600,
-        )
+
+        if config.ENV_DATA.get("cluster_type") != "consumer":
+            # Check for Ceph pods
+            assert pod.wait_for_resource(
+                condition="Running",
+                selector="app=rook-ceph-mon",
```
Comment (ebondare) [inline ocs_ci/deployment/rosa.py:155]: You could use MON_APP_LABEL here

File: ocs_ci/deployment/rosa.py [hunk 2]
```diff
@@ -147,24 +147,29 @@ def deploy_ocs(self):
...
+                resource_count=3,
+                timeout=600,
+            )
+            assert pod.wait_for_resource(
+                condition="Running", selector="app=rook-ceph-mgr", timeout=600
```
Comment (ebondare) [inline ocs_ci/deployment/rosa.py:160]: It would be nice to change the string to constants.MGR_APP_LABEL

File: ocs_ci/deployment/rosa.py [hunk 3]
```diff
@@ -147,24 +147,29 @@ def deploy_ocs(self):
...
+            )
+            assert pod.wait_for_resource(
+                condition="Running",
+                selector="app=rook-ceph-osd",
```
Comment (ebondare) [inline ocs_ci/deployment/rosa.py:164]: The same as above, would be nice to use OSD_APP_LABEL instead

File: ocs_ci/deployment/rosa.py (hunk 1 above)
Comment (jilju) [reply ocs_ci/deployment/rosa.py:155]: Yes but this existing segment of code is not in scope of the requirement addressed in this PR. The change is only adding a condition to run this check.

File: ocs_ci/deployment/rosa.py (hunk 2 above)
Comment (jilju) [reply ocs_ci/deployment/rosa.py:160]: Yes but this existing segment of code is not in scope of the requirement addressed in this PR. The change is only adding a condition to run this check.

File: ocs_ci/deployment/rosa.py (hunk 3 above)
Comment (jilju) [reply ocs_ci/deployment/rosa.py:164]: Yes but this existing segment of code is not in scope of the requirement addressed in this PR. The change is only adding a condition to run this check.

File: ocs_ci/ocs/managedservice.py [hunk 4]
```diff
@@ -78,3 +81,72 @@ def update_pull_secret():
...
+        f'[{{"op": "replace", "path": "/spec/template/spec/containers/0/env", '
+        f'"value":[{{"name": "ROOK_CEPH_USERNAME", "value": "client.admin"}}, '
+        f'{{"name": "ROOK_CEPH_SECRET", "value": "{ceph_admin_key}"}}]}}]'
+    )
+    try:
+        consumer_tools_deployment.patch(params=patch_value, format_type="json")
+    except Exception as exe:
+        logger.warning(
+            "Failed to patch rook-ceph-tools deployment in consumer cluster. "
+            f"The can be done manually after deployment. Error {str(exe)}"
```
Comment (dahorak) [inline ocs_ci/ocs/managedservice.py:139]:
```suggestion
            f"The patch can be done manually after deployment. Error {str(exe)}"
```

Comment (jilju) [reply ocs_ci/ocs/managedservice.py]: Made the changes

File: ocs_ci/deployment/rosa.py (hunk 1 above)
Comment (jilju) [reply ocs_ci/deployment/rosa.py]: Changed to MON_APP_LABEL

File: ocs_ci/deployment/rosa.py (hunk 3 above)
Comment (jilju) [reply ocs_ci/deployment/rosa.py]: Changed to OSD_APP_LABEL

File: ocs_ci/ocs/managedservice.py [hunk 5]
```diff
@@ -93,3 +96,72 @@ def get_consumer_names():
...
+            "Ceph admin key not found to patch rook-ceph-tools deployment on consumer with ceph.admin key. "
+            "Skipping the step."
+        )
+        return
+
+    consumer_tools_pod = get_ceph_tools_pod()
+
+    # Check whether ceph command is working on tools pod. Patch is needed only if the error is "RADOS permission error"
+    try:
+        consumer_tools_pod.exec_ceph_cmd("ceph health")
```
Comment (ebondare) [inline ocs_ci/ocs/managedservice.py:130]: Don't you want to stop the function's execution if the ceph command succeeds?

Comment (jilju) [reply ocs_ci/ocs/managedservice.py:130]: @ebondare Yes, that is correct. Fixed now.

**Review by ebondare — APPROVED** [review]
LGTM

**Review by dahorak — APPROVED** [review]

**Pull request merged**